### ISSUE TITLE
IPV6 support in AWS VPC 

### ISSUE THREAD

**Issue by martin-flaregames** [hashicorp/terraform#11430]
Hi, we can see that there is an implementation for supporting ipv6 in aws_vpc and aws_subnet https://github.com/hashicorp/terraform/pull/10538

But we are not sure if this was already released or not. We are not able to see any relevant configuration in the documentation page: https://www.terraform.io/docs/providers/aws/r/vpc.html

is just the documentation not up to date? or is this feature not yet released? In that case, any time estimation?

Thanks.

### Terraform Version
 `Terraform v0.8.4`

### Affected Resource(s)
- aws_vpc
- aws_subnet

**Comment by stack72** [comment]
Hi @martin-flaregames 

This isn't currently released - it's back on my radar to get working on after Terraform 0.8.5 is released

Thanks and sorry for the time it has taken!

Paul

**Comment by mattlqx** [comment]
AWS just added IPv6 support to VPC and EC2 less than 24 hours ago, unless I missed something. :)

In any case, I'll be excited to see Terraform support IPv6 in AWS resources.

**Comment by unixtastic** [comment]
IPv6 support was released for one region some time back. It got released for all regions yesterday.

I would love to use IPv6 in my VPCs. I'd love to use terraform to configure that.

**Comment by bhuisgen** [comment]
If you add manually an IPv6 CIDR to a VPC, terraform "plan" seems to crash everytime even on on other VPC/regions without IPv6 CIDR. Probably due to the new IPv6 ACL rules ...

```
panic: runtime error: invalid memory address or nil pointer dereference
2017/01/27 23:14:33 [DEBUG] plugin: terraform: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x988197]
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 
2017/01/27 23:14:33 [DEBUG] plugin: terraform: goroutine 1247 [running]:
2017/01/27 23:14:33 [DEBUG] plugin: terraform: panic(0x35717e0, 0xc420010050)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:500 +0x1a1
2017/01/27 23:14:33 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.networkAclEntriesToMapList(0xc42066a680, 0x9, 0x10, 0x32160e0, 0xc42046e360, 0x0)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_network_acl.go:590 +0x257
2017/01/27 23:14:33 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsNetworkAclRead(0xc420401e60, 0x3137960, 0xc42083e1a0, 0x0, 0xf)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_network_acl.go:219 +0x71a
2017/01/27 23:14:33 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc42086b7a0, 0xc42000f270, 0x3137960, 0xc42083e1a0, 0xc420563130, 0x1, 0x18)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0x131
2017/01/27 23:14:33 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4208196e0, 0xc42000f220, 0xc42000f270, 0x0, 0x0, 0x10)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:237 +0x91
2017/01/27 23:14:33 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc420652ea0, 0xc4207c3080, 0xc4208a40b0, 0x0, 0x0)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017/01/27 23:14:33 [DEBUG] plugin: terraform: reflect.Value.call(0xc420014720, 0xc420026058, 0x13, 0x3ca19d0, 0x4, 0xc420765ed0, 0x3, 0x3, 0x2, 0xc4203ac370, ...)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:434 +0x5c8
2017/01/27 23:14:33 [DEBUG] plugin: terraform: reflect.Value.Call(0xc420014720, 0xc420026058, 0x13, 0xc420765ed0, 0x3, 0x3, 0x10fc4bf, 0xc420021f80, 0x1052360)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:302 +0xa4
2017/01/27 23:14:33 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc4208a6100, 0xc4208a60c0, 0xc42089c180, 0xc42038a700, 0xc4205e70a0, 0x3139fe0, 0xc4207c3080, 0x16, 0x313a020, 0xc4208a40b0, ...)
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:383 +0x148
2017/01/27 23:14:33 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2017/01/27 23:14:33 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:477 +0x421
```

**Comment by unixtastic** [comment]
@stack72 - Can you give us an idea when you hope this feature will hit a release?

Not trying to pressure you, a month or two is a totally valid answer and it's not like I'm paying you to do this.. I just really want to use IPv6 on VPC without breaking terraform.

**Comment by martin-flaregames** [comment]
A little off topic, but the ticket from my fellow coworker https://github.com/hashicorp/terraform/issues/11429 is related to this one (Supporting IPV6 in the ALB resource).

**Comment by sean-brandt** [comment]
@stack72 Very interested in the progress on this as well. Have quite a bit of work that this enhancement would make much easier.

**Comment by eternaltyro** [comment]
This is a problem for me as well. I was stupid enough to try enabling IPv6 for one of the (terraform-managed) VPCs and the action created an immutable Network ACL rule for IPv6 for both ingress and egress. 

<img width="778" alt="screen shot 2017-02-24 at 14 34 02" src="https://cloud.githubusercontent.com/assets/230743/23297325/25aa305a-fa9f-11e6-95fa-45162d17c05f.png">

Now terraform crashes always. 

```
2017/02/24 13:53:28 [DEBUG] plugin: terraform: panic: runtime error: invalid memory address or nil pointer dereference
2017/02/24 13:53:28 [DEBUG] plugin: terraform: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x226b71e]
2017/02/24 13:53:28 [DEBUG] plugin: terraform:
2017/02/24 13:53:28 [DEBUG] plugin: terraform: goroutine 2236 [running]:
2017/02/24 13:53:28 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.networkAclEntriesToMapList(0xc4270a4800, 0xf, 0x10, 0x3f7fe20, 0xc4271adbc0, 0x0)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /private/tmp/terraform-20170217-71812-13l2j0q/terraform-0.8.7/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_network_acl.go:590 +0x27e
2017/02/24 13:53:28 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsNetworkAclRead(0xc426c0d980, 0x3e9dc60, 0xc4238d4000, 0x0, 0xf)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /private/tmp/terraform-20170217-71812-13l2j0q/terraform-0.8.7/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_network_acl.go:219 +0x6ec
2017/02/24 13:53:28 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc425ec0000, 0xc426b8ca50, 0x3e9dc60, 0xc4238d4000, 0xc425f29350, 0xc424c36301, 0x6)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /private/tmp/terraform-20170217-71812-13l2j0q/terraform-0.8.7/src/github.com/hashicorp/terraform/helper/schema/resource.go:259 +0x112
2017/02/24 13:53:28 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc425f24120, 0xc426b8ca00, 0xc426b8ca50, 0x7a46c20, 0x0, 0xc426c07e00)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /private/tmp/terraform-20170217-71812-13l2j0q/terraform-0.8.7/src/github.com/hashicorp/terraform/helper/schema/provider.go:237 +0x91
2017/02/24 13:53:28 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc42597cf60, 0xc426c07ea0, 0xc426d61110, 0x0, 0x0)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /private/tmp/terraform-20170217-71812-13l2j0q/terraform-0.8.7/src/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017/02/24 13:53:28 [DEBUG] plugin: terraform: reflect.Value.call(0xc42039da40, 0xc422f43ce8, 0x13, 0x4a609cd, 0x4, 0xc4222e7f20, 0x3, 0x3, 0xc, 0x0, ...)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:434 +0x91f
2017/02/24 13:53:28 [DEBUG] plugin: terraform: reflect.Value.Call(0xc42039da40, 0xc422f43ce8, 0x13, 0xc4204bcf20, 0x3, 0x3, 0x11b9c90, 0x6aad080, 0xc426ca19e0)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /usr/local/Cellar/go/1.8/libexec/src/reflect/value.go:302 +0xa4
2017/02/24 13:53:28 [DEBUG] plugin: terraform: net/rpc.(*service).call(0xc425924cc0, 0xc425924c80, 0xc42599b4f0, 0xc425998b00, 0xc426c85a00, 0x3ea0320, 0xc426c07ea0, 0x16, 0x3ea0360, 0xc426d61110, ...)
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /usr/local/Cellar/go/1.8/libexec/src/net/rpc/server.go:387 +0x144
2017/02/24 13:53:28 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2017/02/24 13:53:28 [DEBUG] plugin: terraform:  /usr/local/Cellar/go/1.8/libexec/src/net/rpc/server.go:481 +0x404
2017/02/24 13:53:28 [ERROR] root.ihs: eval: *terraform.EvalRefresh, err: aws_ecs_cluster.microservice: unexpected EOF
```

```
➜  stage git:(master) ✗ terraform -version
Terraform v0.8.7
```

**Comment by bhuisgen** [comment]
@eternaltyro you can use the plan command option " -target=resource " to bypass the crash on the ACL rules.

**Comment by stack72** [comment]
Closed via #10538
